types/types.go

```diff
@@ -303,3 +306,26 @@ type SystemWideRuleDisable struct {
 	CreatedAt     sql.NullTime `json:"created_at"`
 	UpdatedAT     sql.NullTime `json:"updated_at"`
 }
+
+// ImpactedClustersCnt represents the number of clusters impacted by a rule
+type ImpactedClustersCnt uint32
+
+// RecommendationImpactedClusters is returned by aggregator for the purposes of /rule/ recommendation list endpoint
+type RecommendationImpactedClusters map[RuleID]ImpactedClustersCnt
+
+// RecommendationRow represents a single row in the recommendation table
+type RecommendationRow struct {
+	// RuleID is in "|" format
+	RuleID    RuleID      `json:"rule_id"`
+	RuleFQDN  RuleFQDN    `json:"rule_fqdn"`
+	ErrorKey  ErrorKey    `json:"error_key"`
```

Comment:
It is redundant at the end, you can extract RuleFQDN and ErrorKey from RuleID. What is the reason to keep both?

I would prefer to avoid the RuleID and directly use RuleFQDN and ErrorKey, that are more meaningful for us in our DBs

Reply:
To avoid unnecessary string manipulation + try to optimize everything as much as possible for the new Insights Advisor endpoints, where the full rule_id is the only one ever used. 

Please see https://github.com/RedHatInsights/insights-results-aggregator/pull/1285
https://github.com/RedHatInsights/insights-results-smart-proxy/pull/656
https://github.com/RedHatInsights/insights-results-aggregator/pull/1289

or in the form of a story here :D https://coreos.slack.com/archives/C01A4DTTZN1/p1632987345082000

Reply:
btw, don't we want the `created_at` field too?

Reply:
@epapbak added